artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientSessionImpl.java

```diff
@@ -1901,18 +1901,10 @@ public void rollback(final Xid xid) throws XAException {
          workDone = false;
       } catch (XAException xae) {
          throw xae;
-      } catch (ActiveMQException e) {
-         if (e.getType() == ActiveMQExceptionType.UNBLOCKED || e.getType() == ActiveMQExceptionType.CONNECTION_TIMEDOUT || e.getType() == ActiveMQExceptionType.SHUTDOWN_ERROR) {
-            // Unblocked on failover
-            throw new XAException(XAException.XA_RETRY);
-         }
-
-         // This should never occur
-         XAException xaException = new XAException(XAException.XAER_RMFAIL);
-         xaException.initCause(e);
-         throw xaException;
       } catch (Throwable t) {
-         // This could occur if the TM interrupts the thread
+         if (logger.isTraceEnabled()) {
+            logger.trace("Rollback failed:: {}", convert(xid), t);
+         }
          XAException xaException = new XAException(XAException.XAER_RMFAIL);
```

Comment:
would the TM retry the rollback in such event?

Did this request come from Wildly / @ehsavoie ?

My biggest concern in such changes is the integration with MDBs as I remember having massive headaches with MDB / XA tests back then which I don't want to go back :)

Reply:
@jbertram @ehsavoie I merged this, but this was of course intentional as there was a test validating it. I sent a further commit to fix the test.


So, if you decide against this @ehsavoie please send a revert.